src/components/auth/PasswordSetPage.js

```diff
@@ -0,0 +1,101 @@
+import { Form, Formik } from 'formik'
+import { useEffect } from 'react'
+import { useSelector } from 'react-redux'
+import { Link, Redirect } from 'react-router-dom'
+import { toast } from 'react-toastify'
+import * as Yup from 'yup'
+
+import { resetPassword } from '../../utils/api'
+import { getPathWithMapState } from '../../utils/getInitialUrl'
+import { useAppHistory } from '../../utils/useAppHistory'
+import { PageTemplate } from '../about/PageTemplate'
+import { Input } from '../form/FormikWrappers'
+import Button from '../ui/Button'
+import { Column, FormButtonWrapper, FormInputWrapper } from './AuthWrappers'
+
+const getResetToken = () =>
+  new URLSearchParams(window.location.search).get('token')
+
+const PasswordSetPage = () => {
+  const history = useAppHistory()
+  const { user, isLoading } = useSelector((state) => state.auth)
+
+  useEffect(() => {
+    if (!getResetToken()) {
+      toast.error(
+        "You can't access this page without coming from a password reset email. If you do come from a password reset email, please make sure you used the full URL provided.",
```

Comment:
We should consider showing the error message on the login page after we bounce the user in red text

Reply:
Added TODO comment in #275